File: CSharp/SpecStudioExamplesTestsCSharp/common/PricingTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class PricingTyped
    {
        public Dollar totalPrice;

        public PricingTyped(Dollar totalPrice)
        {
            this.totalPrice = totalPrice;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteString("totalPrice", Json.ToText(this.totalPrice));
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static PricingTyped FromJsonElement(JsonElement m)
        {
            return new PricingTyped(
                new Dollar(Json.AsString(Json.Require(m, "totalPrice"), "totalPrice"))
            );
        }

        public static PricingTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<PricingTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<PricingTyped> FromJSONList(string json)
        {
            var result = new List<PricingTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "PricingTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"TotalPrice={totalPrice}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not PricingTyped other) return false;
            return object.Equals(this.totalPrice, other.totalPrice);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.totalPrice);
            return h.ToHashCode();
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/PricingTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;

public class PricingTyped {
    public Dollar totalPrice;

    public PricingTyped(Dollar totalPrice) {
        this.totalPrice = totalPrice;
    }

    public PricingTyped(PricingString s) {
        this.totalPrice = new Dollar(s.totalPrice);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PricingTyped)) return false;
        PricingTyped that = (PricingTyped) o;
        return Objects.equals(totalPrice, that.totalPrice);
    }

    @Override
    public int hashCode() {
        return Objects.hash(totalPrice);
    }

    public PricingString toPricingString() {
        return new PricingString(String.valueOf(totalPrice));
    }

    public static List<PricingTyped> fromStringList(List<PricingString> list) {
        List<PricingTyped> result = new ArrayList<>();
        for (PricingString s : list) result.add(new PricingTyped(s));
        return result;
    }

    public static List<PricingString> toStringList(List<PricingTyped> list) {
        List<PricingString> result = new ArrayList<>();
        for (PricingTyped t : list) result.add(t.toPricingString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("totalPrice", totalPrice == null ? null : totalPrice.value);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static PricingTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new PricingTyped(
            new Dollar(Json.asString(Json.require(m, "totalPrice"), "totalPrice")));
    }

    public static PricingTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<PricingTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (PricingTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<PricingTyped> fromJSONList(String json) {
        List<PricingTyped> result = new ArrayList<PricingTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "PricingTyped")));
        return result;
    }
}
